machine_learning/similarity_search.py

```diff
@@ -0,0 +1,113 @@
+"""
+Similarity Search : https://en.wikipedia.org/wiki/Similarity_search
+Similarity search is a search algorithm for finding the nearest vector from
+vectors, used in natural language processing.
+In this algorithm, it calculates distance with euclidean distance and
+returns a list containing two data for each vector:
+    1. the nearest vector
+    2. distance between the vector and the nearest vector (float)
+"""
+import math
+
+import numpy as np
+
+
+def euclidean(input_a: np.ndarray, input_b: np.ndarray):
+    """
+    Calculates euclidean distance between two data.
+    :param input_a: ndarray of first vector.
+    :param input_b: ndarray of second vector.
+    :return: Euclidean distance of input_a and input_b. By using math.sqrt(),
+             result will be float.
+
+    >>> euclidean(np.array([0]), np.array([1]))
+    1.0
+    >>> euclidean(np.array([0, 1]), np.array([1, 1]))
+    1.0
+    >>> euclidean(np.array([0, 0, 0]), np.array([0, 0, 1]))
+    1.0
+    """
+
+    dist = 0
+
+    try:
+        for index, v in enumerate(input_a):
+            dist += pow(input_a[index] - input_b[index], 2)
+        return math.sqrt(dist)
+    except TypeError:
+        raise TypeError("Euclidean's input types are not right ...")
+
+
+def similarity_search(dataset: np.ndarray, value: np.ndarray) -> list:
+    """
+    :param dataset: Set containing the vectors. Should be ndarray.
+    :param value: vector/vectors we want to know the nearest vector from dataset.
+    :return: Result will be a list containing
+            1. the nearest vector
+            2. distance from the vector
+
+    >>> a = np.array([[0], [1], [2]])
+    >>> b = np.array([[0]])
+    >>> similarity_search(a, b)
```

Comment:
```suggestion
    >>> dataset = np.array([[0], [1], [2]])
    >>> value_array = np.array([[0]])
    >>> similarity_search(dataset, value_array)
```
Repeat for other these below...

Please add tests that raise errors.